ffmpeg issue?

Recently, I have my Ubuntu changed to 14.04 LTS.
The FFmpeg is not available in this version of linux. so 
$ sudo apt-get install ffmpeg
does not work. 

I can find following.
http://fcorti.com/2014/04/22/ffmpeg-ubuntu-14-04-lts/

Have you ever try to run Pangolin 14.04?

I was trying to compile the file.
make[2]: *** [src/CMakeFiles/pangolin.dir/video/drivers/ffmpeg.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/pangolin.dir/all] Error 2
It seems that the ffmpeg has some problems, and it is hard to install the ffmpeg correctly. BTW, my operating system is ubuntu 14.04 lts. Thanks a lot!

/home/micoarm/Pangolin/src/video/drivers/ffmpeg.cpp:49:39: error: ‘PIX_FMT_YUYV422’ was not declared in this scope
 #define TEST_PIX_FMT_RETURN(fmt) case PIX_FMT_##fmt: return #fmt;
                                       ^
/home/micoarm/Pangolin/src/video/drivers/ffmpeg.cpp:56:5: note: in expansion of macro ‘TEST_PIX_FMT_RETURN’
     TEST_PIX_FMT_RETURN(YUYV422);
     ^
/home/micoarm/Pangolin/src/video/drivers/ffmpeg.cpp:49:39: error: ‘PIX_FMT_RGB24’ was not declared in this scope
 #define TEST_PIX_FMT_RETURN(fmt) case PIX_FMT_##fmt: return #fmt;
                                       ^
/home/micoarm/Pangolin/src/video/drivers/ffmpeg.cpp:57:5: note: in expansion of macro ‘TEST_PIX_FMT_RETURN’
     TEST_PIX_FMT_RETURN(RGB24);
     ^
/home/micoarm/Pangolin/src/video/drivers/ffmpeg.cpp:49:39: error: ‘PIX_FMT_BGR24’ was not declared in this scope
 #define TEST_PIX_FMT_RETURN(fmt) case PIX_FMT_##fmt: return #fmt;
                                       ^
/home/micoarm/Pangolin/src/video/drivers/ffmpeg.cpp:58:5: note: in expansion of macro ‘TEST_PIX_FMT_RETURN’
     TEST_PIX_FMT_RETURN(BGR24);
     ^
/home/micoarm/Pangolin/src/video/drivers/ffmpeg.cpp:49:39: error: ‘PIX_FMT_YUV422P’ was not declared in this scope
 #define TEST_PIX_FMT_RETURN(fmt) case PIX_FMT_##fmt: return #fmt;
                                       ^
/home/micoarm/Pangolin/src/video/drivers/ffmpeg.cpp:59:5: note: in expansion of macro ‘TEST_PIX_FMT_RETURN’
     TEST_PIX_FMT_RETURN(YUV422P);
     ^
/home/micoarm/Pangolin/src/video/drivers/ffmpeg.cpp:49:39: error: ‘PIX_FMT_YUV444P’ was not declared in this scope
 #define TEST_PIX_FMT_RETURN(fmt) case PIX_FMT_##fmt: return #fmt;
                                       ^
/home/micoarm/Pangolin/src/video/drivers/ffmpeg.cpp:60:5: note: in expansion of macro ‘TEST_PIX_FMT_RETURN’
     TEST_PIX_FMT_RETURN(YUV444P);
     ^
/home/micoarm/Pangolin/src/video/drivers/ffmpeg.cpp:49:39: error: ‘PIX_FMT_YUV410P’ was not declared in this scope
 #define TEST_PIX_FMT_RETURN(fmt) case PIX_FMT_##fmt: return #fmt;

Hello stevenlovegrove, here are some error messages. I just clone your latest code on the devel branch. BTW, how to set the cmake cache lines to the empty string by ccmake?
Best regards

--- a/include/pangolin/video/drivers/ffmpeg.h
+++ b/include/pangolin/video/drivers/ffmpeg.h
@@ -91,7 +91,7 @@ protected:
     AVPacket        packet;
     int             numBytesOut;
     uint8_t         *buffer;
-    PixelFormat     fmtout;
+    AVPixelFormat     fmtout;
 };
 
 enum FfmpegMethod
@@ -139,8 +139,8 @@ protected:
     VideoInterface* videoin;
     SwsContext *img_convert_ctx;
     
-    PixelFormat     fmtsrc;
-    PixelFormat     fmtdst;
+    AVPixelFormat     fmtsrc;
+    AVPixelFormat     fmtdst;
     AVFrame*        avsrc;
     AVFrame*        avdst;
     uint8_t*        bufsrc;

--- a/src/video/drivers/ffmpeg.cpp
+++ b/src/video/drivers/ffmpeg.cpp
@@ -37,18 +37,18 @@ extern "C"
 namespace pangolin
 {
 
-PixelFormat FfmpegFmtFromString(const std::string fmt)
+AVPixelFormat FfmpegFmtFromString(const std::string fmt)
 {
     const std::string lfmt = ToLowerCopy(fmt);
     if(!lfmt.compare("gray8") || !lfmt.compare("grey8") || !lfmt.compare("grey")) {
-        return PIX_FMT_GRAY8;
+        return AV_PIX_FMT_GRAY8;
     }
     return av_get_pix_fmt(lfmt.c_str());
 }
 
-#define TEST_PIX_FMT_RETURN(fmt) case PIX_FMT_##fmt: return #fmt;
+#define TEST_PIX_FMT_RETURN(fmt) case AV_PIX_FMT_##fmt: return #fmt;
 
-std::string FfmpegFmtToString(const PixelFormat fmt)
+std::string FfmpegFmtToString(const AVPixelFormat fmt)
 {
     switch( fmt )
     {
@@ -241,7 +241,7 @@ void FfmpegVideo::InitUrl(const std::string url, const std::string strfmtout, co
         throw VideoException("Couldn't allocate frames");
     
     fmtout = FfmpegFmtFromString(strfmtout);
-    if(fmtout == PIX_FMT_NONE )
+    if(fmtout == AV_PIX_FMT_NONE )
         throw VideoException("Output format not recognised",strfmtout);
     
     // Image dimensions
@@ -451,7 +451,7 @@ bool FfmpegConverter::GrabNewest( unsigned char* image, bool wait )
 
 // Based on this example
 // http://cekirdek.pardus.org.tr/~ismail/ffmpeg-docs/output-example_8c-source.html
-static AVStream* CreateStream(AVFormatContext *oc, CodecID codec_id, uint64_t frame_rate, int bit_rate, PixelFormat EncoderFormat, int width, int height)
+static AVStream* CreateStream(AVFormatContext *oc, CodecID codec_id, uint64_t frame_rate, int bit_rate, AVPixelFormat EncoderFormat, int width, int height)
 {
     AVCodec* codec = avcodec_find_encoder(codec_id);
     if (!(codec)) throw
@@ -518,8 +518,8 @@ protected:
     FfmpegVideoOutput& recorder;
 
     StreamInfo input_info;
-    PixelFormat input_format;
-    PixelFormat output_format;
+    AVPixelFormat input_format;
+    AVPixelFormat output_format;
 
     AVPicture src_picture;
     AVPicture dst_picture;

